core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -381,31 +381,48 @@ object SQLPlanParser extends Logging {
     funcName
   }
 
-  private def getAllFunctionNames(functionPattern: Regex, expr: String,
+  // This method aims at doing some common processing to an expression before
+  // we start parsing it. For example, some special handling is required for some functions.
+  private def processSpecialFunctions(expr: String): String = {
+    // For parse_url, we only support parse_url(*,Host,*); parse_url(*,Protocol,*)
+    // So we want to be able to define that parse_url(*,QUERY,*) is not supported.
+
+    // The following regex uses forward references to find matches for parse_url(*)
+    // we need to use forward references because otherwise multiple occurrences will be be matched
+    // only once.
+    // https://stackoverflow.com/questions/47162098/is-it-possible-to-match-nested-brackets-with-a-
+    // regex-without-using-recursion-or/47162099#47162099
+    // example parse_url:
+    // Project [url_col#7, parse_url(url_col#7, HOST, false) AS HOST#9,
+    //          parse_url(url_col#7, QUERY, false) AS QUERY#10]
+    val parseURLPattern = ("parse_url(?=\\()(?:(?=.*?\\((?!.*?\\1)(.*\\)(?!.*\\2).*))(?=.*?\\)" +
+      "(?!.*?\\2)(.*)).)+?.*?(?=\\1)[^(]*(?=\\2$)").r
+    var newExpr = expr
+    parseURLPattern.findAllMatchIn(expr).foreach { parse_call =>
+      // iterate on all matches replacing parse_url by parse_url_query
+      // note that we do replaceFirst because we want to  map 1-to-1 and the order does
```

Comment:
nit: extra blank space between "to  map"

Reply:
done!